hack/verify-structured-logging.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to avoid regressions after a package is migrated
+# to structured logging. once a package is completely migrated add
+# it .structured_logging file to avoid any future regressions.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+migrated_packages_file="${KUBE_ROOT}/hack/.structured_logging"
+kube::util::check-file-in-alphabetical-order "${migrated_packages_file}"
+
+migrated_packages=()
+while IFS='' read -r line; do
+  migrated_packages+=("$line")
+done < <(cat "${migrated_packages_file}")
+
+# Install logcheck
+pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+  GO111MODULE=on go install k8s.io/klog/hack/tools/logcheck
+popd >/dev/null
+
+# TODO: Improve concurrancy here
+for package in "${migrated_packages[@]}"; do
+  logcheck "$KUBE_ROOT/$package" || continue
```

Comment:
why is this `|| continue` ? there's nothing else in the loop so we're not continuing past anything but doing `|| continue` will mask the error if there is a failing package?

Reply:
```sh
ret=0
for package in "${migrated_packages[@]}"; do
  logcheck "$KUBE_ROOT/$package" || ret=$?
done
exit $ret
```
(and possibly check ret and print a message)

Reply:
Is the ret stuff necessary? It will exit on that return code without this. (Maybe shellcheck was complaining?)

Reply:
we could drop the `||continue` and it will exit on the first one, but IMHO it's a better experience to log all errors and then exit failure (so you can fix all of them instead of only seeing the first one we hit until you fix that), using the ret thing lets us retain if any of them failed while running all of them.

Reply:
Done.